tests/trx_generator/trx_generator.cpp

```diff
@@ -167,7 +156,69 @@ namespace eosio::testing {
       return true;
    }
 
-   bool transfer_trx_generator::tear_down() {
+   fc::variant json_from_file_or_string(const string& file_or_str, fc::json::parse_type ptype = fc::json::parse_type::legacy_parser)
+   {
+      regex r("^[ \t]*[\{\[]");
+      if ( !regex_search(file_or_str, r) && fc::is_regular_file(file_or_str) ) {
+         try {
+            return fc::json::from_file(file_or_str, ptype);
+         } EOS_RETHROW_EXCEPTIONS(json_parse_exception, "Fail to parse JSON from file: ${file}", ("file", file_or_str));
+
+      } else {
+         try {
+            return fc::json::from_string(file_or_str, ptype);
+         } EOS_RETHROW_EXCEPTIONS(json_parse_exception, "Fail to parse JSON from string: ${string}", ("string", file_or_str));
+      }
+   }
+
+   trx_generator::trx_generator(std::string chain_id_in, const std::string& abi_data_file, std::string contract_owner_account, std::string auth_account, std::string action_name,
+         const std::string& action_data_file_or_str, fc::microseconds trx_expr, const std::string& private_key_str, std::string lib_id_str, std::string log_dir)
+       : trx_generator_base(chain_id_in, contract_owner_account, trx_expr, lib_id_str, log_dir), _abi_data_file_path(abi_data_file), _auth_account(auth_account),
+         _action(action_name), _action_data_file_or_str(action_data_file_or_str), _private_key(fc::crypto::private_key(private_key_str)) {}
+
+   bool trx_generator::setup() {
+      _nonce_prefix = 0;
+      _nonce = static_cast<uint64_t>(fc::time_point::now().sec_since_epoch()) << 32;
+
+      std::cout
+            << "Stop Generation (form potential ongoing generation in preparation for starting new generation run)."
+            << std::endl;
```

Comment:
This can be a separate PR if you like. Most of trx_generator uses `xlog()` statements instead of `cout`.

Reply:
Good call.  Addressed here: [d867824](https://github.com/AntelopeIO/leap/pull/626/commits/d86782461b5914287da0e9b2ac48689013a155c4)